### PR TITLE
feat: add shady-link warning

### LINKED ISSUE #80: Detect shady links
My idea of this task is to implement a new warning responsible of detecting shady links in Literals. I have been inspired by one of the detection of [guarddog](https://github.com/DataDog/guarddog) from DataDog.

They use the following RegEx:
`(http[s]?:\/\/.*\.(link|xyz|tk|ml|ga|cf|gq|pw|top|club|mw|bd|ke|am|sbs|date|quest|cd|bid|cd|ws|icu|cam|uno|email|stream))`

This RegEx allow to detect an URL like that:

```
https://foobar.xyz
```

The idea is to create a new probe or to update isLiteral probe to add that new warning. The probe will execute on every ESTree Literal.

```
const maliciousUrl = "https://foobar.xyz";
```

---

Maybe we need to conduct additional research on the subject (maybe there is some study we may want to read to improve the detection?).

### PR DESCRIPTION
**Pull request by PierreDemailly**
This PR fixes #80

### CODE REVIEW

**Review by antoine-coulon — APPROVED** [review]
Nice one! LGTM

File: src/probes/isLiteral.js [hunk 1]
```diff
@@ -37,6 +37,9 @@ function main(node, options) {
       analysis.addWarning("encoded-literal", node.value, node.loc);
     }
   }
+  else if (/(http[s]?:\/\/.*\.(link|xyz|tk|ml|ga|cf|gq|pw|top|club|mw|bd|ke|am|sbs|date|quest|cd|bid|cd|ws|icu|cam|uno|email|stream))$/.test(node.value)) {
```
Comment (Kawacrepe) [inline src/probes/isLiteral.js]: If I understand correctly, on guarddog they also consider `(http[s]?:\/\/bit\.ly.*)$` as a shady link regex, shouldn't we add it?

Comment (tony-go) [inline src/probes/isLiteral.js]: We could put the regex into a variable. It could improve readability.

**Review by tony-go — APPROVED** [review]
lgtm & my comment

File: src/probes/isLiteral.js (hunk 1 above)
Comment (PierreDemailly) [reply src/probes/isLiteral.js]: I think we should ;) It's done

Comment (PierreDemailly) [reply src/probes/isLiteral.js]: I agree, it's done

**Review by Kawacrepe — APPROVED** [review]

**Review by fraxken — APPROVED** [review]
LGTM

**Comment by fraxken** [comment]
@all-contributors please add @PierreDemailly for code, test

**Comment by cccs-kevin** [comment]
This is awesome, I look forward to the next release that includes this!

**Pull request merged**